Challenge3.js

```diff
@@ -0,0 +1,55 @@
+// Write a function `serialPromise()` that will execute promises in series.
+
+// Example promise function:
+// ```js
+function delayPromise(ms) {
+    return new Promise(resolve => {
+        setTimeout(() => {
+            console.log('delayPromise(%d) done', ms);
+            resolve(2*ms);
+        }, ms);
+    });
+}
+// ```
+// I should be able to run:
+
+// ```js
+// console.time('serialPromise')
+// serialPromise(delayPromise, [100,200,700,1000]).then(() => console.timeEnd('serialPromise'));
+// // ```
+// Resulting in something like:
+// ```
+// delayPromise(100) done
+// delayPromise(200) done
+// delayPromise(700) done
+// delayPromise(1000) done
+// serialPromise: 2002.3681640625ms
+// ```
+
+// ### Bonus
+// Have `serialPromise()` resolve to an array of the individual promise resolutions:
+
+// ```js
+// serialPromise(delayPromise, [100,500,100,600]).then(console.log);
+// ```
+
+// Would result in:
+// ```js
+// [200, 1000, 200, 1200]
+// ```
+
+// Because of the `resolve(2*ms)` in the `delayPromise()` function.
+
+function serialPromise(func, arr) {
```

Comment:
This is actually running the promises in parallel with `Promise.all()`.

Timing it shows that it takes about 1000ms (the longest single task) instead of the 2000ms it should take if they were done one at a time:

```js
console.time('serialPromise');
serialPromise(delayPromise, [100,200,700,1000]).then(() => console.timeEnd('serialPromise'));
```